middle_end/flambda2/simplify_shared/slot_offsets.ml

```diff
@@ -853,15 +889,22 @@ end = struct
     state.sets_of_closures <- set :: state.sets_of_closures;
     (* Fill closure slots *)
     Function_slot.Map.iter
-      (fun function_slot code_id ->
+      (fun function_slot
+           (code_id : Function_declarations.code_id_in_function_declaration) ->
+        let code_id =
+          match code_id with Deleted -> None | Code_id code_id -> Some code_id
+        in
         let s =
           match
             Function_slot.Map.find_opt function_slot state.function_slots
           with
           | None ->
             create_function_slot set state get_code_metadata function_slot
               code_id
-          | Some s -> s
+          | Some s ->
+            s.sets <- set :: s.sets;
+            update_set_for_slot s set;
+            s
```

Comment:
Could you explain what this patch does (and the same patches later for value slots) ? At first glance it looks unrelated to the rest of the PR

Reply:
It's indeed unrelated to the rest of the PR, and is a fix for a "how-did-it-ever-work-before" kind of bug: previously, slots were only considered to be part of the first set of closures they were seen in. Currently, it works because two set of closures which have a slot in common have exactly the same slots. Maybe I should open another PR with those changes?

Reply:
Do you think you could write an fexpr program that tests this ? We can also wait for the global pass, which will make it easier to write such programs in OCaml directly.

Reply:
Now merged as a separate PR.